dask_expr/_collection.py

```diff
@@ -2555,6 +2555,25 @@ def from_map(
         # This option doesn't really make sense in dask-expr
         raise NotImplementedError("dask_expr does not support a token argument.")
 
+    lengths = set()
+    iterables = list(iterables)
+    for i, iterable in enumerate(iterables):
+        if not isinstance(iterable, Iterable):
+            raise ValueError(
+                f"All elements of `iterables` must be Iterable, got {type(iterable)}"
+            )
+        try:
+            lengths.add(len(iterable))
+        except (AttributeError, TypeError):
+            iterables[i] = list(iterable)
+            lengths.add(len(iterables[i]))
+    if len(lengths) == 0:
+        raise ValueError("`from_map` requires at least one Iterable input")
+    elif len(lengths) > 1:
+        raise ValueError("All `iterables` must have the same length")
+    if lengths == {0}:
+        raise ValueError("All `iterables` must have a non-zero length")
```

Comment:
Couldn't this be `lengths = {0, 1}` and thus still need to fail?
```suggestion
    if 0 in lengths:
        raise ValueError("All `iterables` must have a non-zero length")
```

Reply:
That is covered by the previous condition that raises if you have more than one length